TriangleApp/app/src/main/java/triangle/triangleapp/helpers/MediaHelper.java

```diff
@@ -0,0 +1,192 @@
+package triangle.triangleapp.helpers;
+
+import android.hardware.Camera;
+import android.media.CamcorderProfile;
+import android.media.MediaRecorder;
+import android.os.Environment;
+import android.util.Log;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import triangle.triangleapp.view.CameraPreview;
+
+/**
+ * Created by danie on 9-6-2017.
+ */
+
+public class MediaHelper {
+    private final String TAG = "MediaHelper";
+    private boolean isRecording;
+    private MediaRecorder mMediaRecorder;
+    private CameraPreview mCameraPreview;
+    private Camera mCamera;
+    private final int MEDIA_TYPE_IMAGE = 1;
+    private final int MEDIA_TYPE_VIDEO = 2;
+    private WebSocket webSocket;
+    private String url;
+    private String protocol;
+    private int mediaRecorderMaxDuration;
+
+    public MediaHelper(CameraPreview cameraPreview) {
+        mCameraPreview = cameraPreview;
+        isRecording = false;
+        mediaRecorderMaxDuration = 5000;
+        url = "ws://145.49.35.215:1234/send";
+        protocol = "ws";
+        webSocket = new WebSocket(url, protocol);
+    }
+
+    public void record(){
+        if (isRecording) {
+            isRecording = false;
+            stopStreaming(true);
+        } else {
+            startStreaming(true);
+        }
+    }
+
+    private void initializeVideoRecorder(boolean firstInit) {
+        if (firstInit) {
+            mCamera = CameraHelper.getCameraInstance();
+            mMediaRecorder = new MediaRecorder();
+
+            // Step 1: Unlock and set camera to MediaRecorder
+            mCamera.unlock();
+            mMediaRecorder.setCamera(mCamera);
+        }
+
+        // Step 2: Set sources
+        mMediaRecorder.setAudioSource(MediaRecorder.AudioSource.CAMCORDER);
+        mMediaRecorder.setVideoSource(MediaRecorder.VideoSource.CAMERA);
+
+        // Step 3: Set a CamcorderProfile (requires API Level 8 or higher)
+        mMediaRecorder.setProfile(CamcorderProfile.get(CamcorderProfile.QUALITY_480P));
+
+        // Step 4: Set output file
+        final String fileName = getOutputMediaFile(MEDIA_TYPE_VIDEO).toString();
+        mMediaRecorder.setOutputFile(fileName);
+
+        // Step 5: Set the preview output
+        mMediaRecorder.setPreviewDisplay(mCameraPreview.getHolder().getSurface());
+
+        mMediaRecorder.setOnInfoListener(new MediaRecorder.OnInfoListener() {
+            @Override
+            public void onInfo(MediaRecorder mr, int what, int extra) {
+                // Handle the on duration exceeded
+                if (what == MediaRecorder.MEDIA_RECORDER_INFO_MAX_DURATION_REACHED) {
+                    stopStreaming(false);
+                    startStreaming(true);
+
+                    if (webSocket.isConnected()) {
```

Comment:
Created new class: MediaFileHelper, Replaced methods: getOutputMediaFile and getBytesFromFile to new class. Added exception handling in methods.